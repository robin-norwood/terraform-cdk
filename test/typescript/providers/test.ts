import { TestDriver } from "../../test-helper";

describe("full integration test", () => {
  let driver: TestDriver;

  beforeAll(async () => {
    driver = new TestDriver(__dirname);
    await driver.setupTypescriptProject();
  });

  test("build providers", () => {
    expect(driver.deploy("using-all-providers")).toMatchSnapshot();
  });

  describe("provider functionality", () => {
    beforeAll(async () => {
      await driver.synth();
    });

    test("namespaces", () => {
      const stack = driver.synthesizedStack("namespaces");
      expect(stack).toMatchSnapshot();
    });

    test("references", () => {
      const stack = driver.synthesizedStack("references");
      // TODO: write a proper test matcher
      expect(stack).toMatchSnapshot();
    });

    test("mutation", () => {
      const stack = driver.synthesizedStack("mutation");
      // TODO: write a proper test matcher
      expect(stack).toMatchSnapshot();
    });
  });
});
