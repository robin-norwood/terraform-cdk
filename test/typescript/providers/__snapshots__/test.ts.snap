// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full integration test build providers 1`] = `
"Deploying Stack: hello-deploy
Resources
 ✔ NULL_RESOURCE       test                null_resource.test


Summary: 1 created, 0 updated, 0 destroyed.
"
`;

exports[`full integration test provider functionality references 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"references\\",
      \\"backend\\": \\"local\\"
    }
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"null\\": {
        \\"version\\": \\" ~> 3.1.0\\",
        \\"source\\": \\"null\\"
      }
    }
  },
  \\"provider\\": {
    \\"null\\": [
      {}
    ]
  },
  \\"resource\\": {
    \\"null_resource\\": {
      \\"test\\": {
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"references/test\\",
            \\"uniqueId\\": \\"test\\"
          }
        }
      }
    }
  }
}"
`;
